website/content/docs/agent/index.mdx

```diff
@@ -8,18 +8,94 @@ description: |-
 
-Vault Agent is a client daemon that provides the following features:
+Nearly all requests to Vault must be accompanied by a valid client token. This
+includes all API requests, as well as via the Vault CLI and other libraries.
+Therefore, Vault clients must first authenticate with Vault to acquire a token,
+and Vault provides a number of different authentication methods to assist in the
+delivery of this initial token.
+
+![Client authentication](/img/diagram-vault-agent.png)
+
+If the client can securely get the token, all subsequent requests (e.g. request
+database credentials, read key/value secrets) can be processed based on the
+trust established by successful authentication.
+
+This means that the client application must invoke the Vault API to authenticate
+with Vault and manage the acquired token in addition to invoking the API to
+request secrets from Vault. This implies code changes to client applications
+along with additional testing and maintenance of the application.
+
+The following code example implements Vault API to authenticate with Vault
+through [AppRole auth method](/docs/auth/approle#code-example), and then uses
+the returned client token to read secrets at `kv-v2/data/creds`.
+
+```go
+package main
+
+import (
+    ...snip...
+    vault "github.com/hashicorp/vault/api"
+)
+
+// Fetches a key-value secret (kv-v2) after authenticating via AppRole
+func getSecretWithAppRole() (string, error) {
+    config := vault.DefaultConfig() 
+
+    client := vault.NewClient(config)
+    wrappingToken := ioutil.ReadFile("path/to/wrapping-token") 
+    unwrappedToken := client.Logical().Unwrap(strings.TrimSuffix(string(wrappingToken), "\n"))
+   
+    secretID := unwrappedToken.Data["secret_id"]
+    roleID := os.Getenv("APPROLE_ROLE_ID")
+
+    params := map[string]interface{}{
+        "role_id":   roleID,
+        "secret_id": secretID,
+    }
+    resp := client.Logical().Write("auth/approle/login", params)
+    client.SetToken(resp.Auth.ClientToken)
 
-- [Auto-Auth][autoauth] - Automatically authenticate to Vault and manage the token renewal process for locally-retrieved dynamic secrets.
-- [Caching][caching] - Allows client-side caching of responses containing newly created tokens and responses containing leased secrets generated off of these newly created tokens.
-- [Windows Service][winsvc] - Allows running the Vault Agent as a Windows service.
-- [Templating][template] - Allows rendering of user supplied templates by Vault Agent, using the token generated by the Auto-Auth step.
-  To get help, run:
+    secret, err := client.Logical().Read("kv-v2/data/creds")
+    if err != nil {
+        return "", fmt.Errorf("unable to read secret: %w", err)
+    }
 
-```shell-session
-$ vault agent -h
+    data := secret.Data["data"].(map[string]interface{})
+    
+    ...snip...
+}
 ```
 
+For some Vault deployments, making (and maintaining) these changes to
+applications may not be a problem, and may actually be preferred. This could be
```

Comment:
Just a suggestion but you could also cut this sentence short by saying: For some Vault deployments, making (and maintaining) these changes to applications is preferred.

Unless, of course, you need to stress that these changes to the applications aren't a problem.